server/controllers/contest.js

```diff
@@ -0,0 +1,58 @@
+const Contest = require("../models/Contest");
+const asyncHandler = require("express-async-handler");
+
+exports.createContest = asyncHandler(async (req, res) => {
+    try {
+        // expecting req.body to come back in this format.
```

Comment:
Please include the `protect` middleware to your routes, to protect them and access the user's id using the request object, `req.user.id`

Reply:
perfect, should have seen that. Will update shortly